fastly/image_optimizer_default_settings.go

```diff
@@ -0,0 +1,187 @@
+package fastly
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// ResizeFilter is a base for the different ResizeFilter variants.
+type ResizeFilter int64
+
+func (r ResizeFilter) String() string {
+	switch r {
+	case Lanczos3:
+		return "lanczos3"
+	case Lanczos2:
+		return "lanczos2"
+	case Bicubic:
+		return "bicubic"
+	case Bilinear:
+		return "bilinear"
+	case Nearest:
+		return "nearest"
+	}
+	return "lanczos3" // default
+}
+
+func (r ResizeFilter) MarshalJSON() ([]byte, error) {
+	return json.Marshal(r.String())
+}
+
+const (
+	// A Lanczos filter with a kernel size of 3. Lanczos filters can detect edges and linear features within an image, providing the best possible reconstruction.
+	Lanczos3 ResizeFilter = iota
+	// A Lanczos filter with a kernel size of 2.
+	Lanczos2
+	// A filter using an average of a 4x4 environment of pixels, weighing the innermost pixels higher.
+	Bicubic
+	// A filter using an average of a 2x2 environment of pixels.
+	Bilinear
+	// A filter using the value of nearby translated pixel values. Preserves hard edges.
+	Nearest
+)
+
+// JpegType is a base for different JpegType variants
+type JpegType int64
```

Comment:
This custom type should be prefixed with `ImageOpto` as it will be globally available across the whole go-fastly package and in the future there might be other features/APIs that have things related to jpegs (maybe?), so we need to be able to distinguish this type as being related to ImageOpto.

> **EDIT:** Let's consider this a general rule across this whole file as I realised there are a few different consts and types that could overlap with future requirements.